internal/internal_worker.go

```diff
@@ -953,6 +994,14 @@ func (aw *AggregatedWorker) Start() error {
 	}
 	proto.Merge(aw.capabilities, capabilities)
 
+	return aw.memoizedStart()
```

Comment:
I think this whole method should go inside memoized start. No need to repeat stuff above for each call.

Reply:
Works for me, this was something that @Quinn-With-Two-Ns requested, so just confirming he's also okay with that.

Reply:
Users may retry starting their worker if it failed that is why I requested we don't memorize it, I think the likely hood is low, but it's very little effort to not memorize it so why not just avoid the breaking change?

Reply:
I changed it to what @cretz suggested, I'm open to changing back.
I don't have a strong opinion here but slightly prefer memoizing the entire thing because it's easier to reason about.